### PR TITLE
[NO ISSUE] Update What's New and release notes for Analytics

### PR DESCRIPTION
**Pull request by simon-dew**
The following draft documentation is ready for review:

* [What's New](https://github.com/couchbase/docs-server/blob/e2b0403af160973978f2e2641a46164f6dee92e9/modules/introduction/pages/whats-new.adoc) — added Analytics features
* [Release notes](https://github.com/couchbase/docs-server/blob/e2b0403af160973978f2e2641a46164f6dee92e9/modules/release-notes/pages/relnotes.adoc) — added Analytics features and enhancements

### CODE REVIEW

**Comment by westmann** [comment]
The order of the Analytics features should be:
- HA
- Views
- Tableau
- Array Indexes
- Azure

(Both for the "What’s New in Version 7.1 "and for the "Release Notes for Couchbase Server 7.1".)

**Comment by idris-motiwala** [comment]
We should term "Views" as "Tabular Analytics Views" as "Views" could get confused with existing Views on our platform which are going to be deprecated. From the order of analytics features, I agree with Till's comment but would like to promote Azure Blob Storage before Array Indexes. Also, could we combine Tabular Analytics Views with the Tableau Connector section as both are connected features.

**Comment by westmann** [comment]
We support 2 different kinds of Analytics Views ("regular" views and tabular views) and I think that the current description covers this well. I also think that we should separate the Tableau Connector from the Analytics views as - while they are indeed connected - they are independently usable.

File: modules/introduction/pages/whats-new.adoc [hunk 1]
```diff
@@ -12,12 +12,60 @@ For information about platform support changes, deprecation notifications, notab
...
+Refer to xref:analytics:rest-admin.adoc#cluster-status[Cluster Status].
+
+=== Analytics Service: Views
+
+Analytics now supports _views_, which are virtual Analytics collections derived from actual stored Analytics collections.
+Analytics views might be used to present a subset of the stored data, or to present a join or other transformation of stored data for some particular purpose.
+Queries may operate on Analytics views in exactly the same way as they do on real stored data.
+
+Analytics also supports _tabular views_, a type of view that presents data in the form of a table with uniform rows and columns and with a well-defined schema.
+Tabular views enables Analytics to interact with software tools that are designed for relational databases, such as Tableau.
```
Comment (RayOffiah) [inline modules/introduction/pages/whats-new.adoc]:
```suggestion
Tabular views enables Analytics to interact with software tools that are designed for relational databases, such as https://www.tableau.com[Tableau].
```

File: modules/introduction/pages/whats-new.adoc [hunk 2]
```diff
@@ -12,12 +12,60 @@ For information about platform support changes, deprecation notifications, notab
...
+Refer to xref:tableau-connector::index.adoc[].
+
+=== Analytics Service: Array Indexes
+
+Analytics now provides _array indexes_, which enable you to index values within an array, or fields within an object nested in an array.
+In Analytics, array indexes are not meant to serve as covering indexes.
+Instead, array indexes are simply meant to accelerate queries involving multi-valued fields.
+For further details, refer to xref:analytics:7_using_index.adoc[].
+
+The data definition language has been extended to support creating and dropping of array indexes.
```
Comment (RayOffiah) [inline modules/introduction/pages/whats-new.adoc]:
```suggestion
The data definition language has been extended to support the creating and dropping of array indexes.
```

**Review by RayOffiah — APPROVED** [review]
Approved with minor suggestions.

File: modules/introduction/pages/whats-new.adoc (hunk 1 above)
Comment (simon-dew) [reply modules/introduction/pages/whats-new.adoc]: Done

File: modules/introduction/pages/whats-new.adoc (hunk 2 above)
Comment (simon-dew) [reply modules/introduction/pages/whats-new.adoc]: Done

**Comment by simon-dew** [comment]
> The order of the Analytics features should be:

Done

**Comment by simon-dew** [comment]
> We should term "Views" as "Tabular Analytics Views" as "Views" could get confused with existing Views on our platform which are going to be deprecated.

Reworded as "Analytics Views" and "Tabular Analytics Views" to avoid confusion.

> From the order of analytics features, I agree with Till's comment but would like to promote Azure Blob Storage before Array Indexes.

Done

> Also, could we combine Tabular Analytics Views with the Tableau Connector section as both are connected features.

Per @westmann 's comments, I've kept these as separate points, but they are both clearly related to each other in the running text.

**Pull request merged**